cluster-autoscaler/cloudprovider/azure/azure_agent_pool.go

```diff
@@ -404,6 +465,16 @@ func (as *AgentPool) DeleteNodes(nodes []*apiv1.Node) error {
 		refs = append(refs, ref)
 	}
 
+	succeededAndFailedDeployments, err := as.getAllSucceededAndFailedDeployments()
+	if err != nil {
+		return err
+	}
+
+	err = as.deleteOutdatedDeployments(succeededAndFailedDeployments)
+	if err != nil {
+		return err
```

Comment:
same here, log a warning and continue the next steps, cleanup job shouldn't be blocking